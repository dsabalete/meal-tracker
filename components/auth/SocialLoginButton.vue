<template>
  <button
    class="rounded-full border py-2 px-6 flex items-center"
    :class="type"
    @click="$emit('click')"
  >
    <component :is="socialComponent" class="w-6 mr-2" />
    {{ text }}
  </button>
</template>

<script>
export default {
  name: 'AuthSocialLoginButton',
  props: {
    type: {
      type: String,
      default: 'google',
      validate: (value) => ['google', 'github'].includes(value),
    },
    text: {
      type: String,
      required: true,
    },
  },
  data: () => ({
    socialOptions: {
      google: 'IconGoogle',
      github: 'IconGithub',
    },
  }),
  computed: {
    socialComponent() {
      return this.socialOptions[this.type]
    },
  },
}
</script>

<style lang="scss" scoped>
.google {
  @apply bg-red-main-600 border-red-main-600 text-white;
}
.github {
  @apply bg-gray-main-1000  border-white text-white;
}
</style>
