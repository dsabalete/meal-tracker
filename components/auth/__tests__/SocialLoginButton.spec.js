import { mount } from '@vue/test-utils'
import SocialLoginButton from '@/components/auth/SocialLoginButton.vue'

const build = (props) =>
  mount(SocialLoginButton, {
    propsData: {
      ...props,
    },
  })

describe('SocialLoginButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = build({
      text: 'asdf',
    })
  })

  test('should be render', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
