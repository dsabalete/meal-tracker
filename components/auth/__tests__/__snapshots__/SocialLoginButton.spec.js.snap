// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginButton should match snapshot 1`] = `
<button
  class="rounded-full border py-2 px-6 flex items-center google"
>
  <icongoogle
    class="w-6 mr-2"
  />
  
  asdf

</button>
`;
